demos/classification_benchmark_demo/cpp/main.cpp

```diff
@@ -1,35 +1,37 @@
-// Copyright (C) 2018-2019 Intel Corporation
+// Copyright (C) 2020-2022 Intel Corporation
 // SPDX-License-Identifier: Apache-2.0
 //
 
-#include <vector>
-#include <queue>
-#include <memory>
-#include <string>
 #include <chrono>
-#include <condition_variable>
-#include <mutex>
-#include <cstdio>
-#include <functional>
-#include <atomic>
+#include <fstream>
+#include <iostream>
+#include <limits>
+#include <string>
+#include <vector>
 
-#include <inference_engine.hpp>
 #include <gflags/gflags.h>
+#include <opencv2/opencv.hpp>
+#include <openvino/openvino.hpp>
+
+#include <models/classification_model.h>
+#include <models/results.h>
 #include <pipelines/async_pipeline.h>
 #include <pipelines/metadata.h>
-#include <models/classification_model.h>
-#include <utils/common.hpp>
-#include <utils/slog.hpp>
+
 #include <utils/args_helper.hpp>
+#include <utils/common.hpp>
 #include <utils/ocv_common.hpp>
 #include <utils/performance_metrics.hpp>
+#include <utils/slog.hpp>
 
 #include "grid_mat.hpp"
 
 static const char help_message[] = "Print a usage message.";
 static const char image_message[] = "Required. Path to a folder with images or path to an image file.";
 static const char model_message[] = "Required. Path to an .xml file with a trained model.";
 static const char labels_message[] = "Required. Path to .txt file with labels.";
+static const char layout_message[] = "Optional. Model inputs layouts. "
+                                     "Ex. \"[NCHW]\" or \"input1[NCHW],input2[NC]\" in case of more than one input.";
```

Comment:
You should mimic the syntax of `--nstreams`

Reply:
I am not sure about it, `--nstreams` are passed as `<device1>:<nstreams1>,<device2>:<nstreams2>` and `:` is used as delimiter here because devices don't have it in names. `--layout` flag use `[` `]` as delemeters, and  because tensor inputs might have `:` in names it's not reasonable to mimic `--nstreams`.

Reply:
I would suggest using same syntax everywhere, but @ivikhrev is right, we cannot use : as delimeter. What's about changing other parameters to [] syntax ?

Reply:
There's still a way to keep `--nstreams` syntax:
Find last `:`
If there exits one, everything after this `:` is a layout
Find last `,`
If there exits one, everything after this `,` but before found `:` on prev step is a name

Reply:
The current target is to have the patch merged by tomorrow. You can ignore this conversations for the sake of speed

Reply:
I already started to change it along with returning auto_resize.

Reply:
Since python PR was already merged without parsing update, I will keep this changes for the future patch. I'll concentrate on returning auto_resize now, it shouldn't take too long.

Reply:
Strange, from my perspective I posted
> FYI, I've merged https://github.com/openvinotoolkit/open_model_zoo/pull/3080

when there was no you answer.

Ok then